app/assets/javascripts/services/PostChatSurveyDigitalAssistantService.js

```diff
@@ -0,0 +1,246 @@
+            siteID: Number(chatParams.siteID),
+            customerID: chatParams.thisCustomerID,
+            incAssignmentID: chatParams.sessionID,
+            pageID: Number(chatParams.launchPageId),
+            sessionID: chatParams.sessionID,
+            chatID: chatParams.chatID,
+            agentID: chatParams.agentID,
+            automatonName: automaton.name,
+            preAssigned: this.sdk.isConnected() && !chatParams.agentID,
+            automatonID: automaton.id,
+            custID: chatParams.thisCustomerID,
+            visitorAttributes: chatParams.getVisitorAttributes(),
+            automatonAttributes: "",
+            clientID: Number(chatParams.siteID),
+            businessUnitID: chatParams.businessUnitID,
+            businessRuleID: Number(chatParams.brID),
+            busUnitID: chatParams.businessUnitID,
+            BRName: chatParams.chatTitle,
+            agentGroupID: chatParams.agId,
+            availableAgentAttributes: chatParams.agentAttributes,
+            brAttributes: chatParams.ruleAttributes,
+            countryCode: chatParams.countryCode,
+            regionCode: chatParams.regionCode,
+            deviceType: chatParams.deviceType,
+            operatingSystemType: chatParams.operatingSystemType,
+            browserType: chatParams.browserType,
+            browserVersion: chatParams.browserVersion,
+            surveyId: Number(survey.id),
+            "automaton.outcomeType": "Completed",
+            "automaton.outcome": "User has submitted postchat feedback.",
+            clientTimestamp: timestamp,
+            automatonOrigin: "richMedia"
+        };
+        console.log("== submitPostChatSurvey ==");
+
+        try {
+            this.sdk.logEventToDW({ eventList: [customerRespondedEvent, endedEvent] });
+        } catch (e) {
+            console.error("!!!! logEventToDW got exception: ", e);
+        }
+    }
+
+    closePostChatSurvey(automaton, timestamp) {
+        const chatParams = this.sdk.getChatParams();
+
+        const endedEvent = {
+            _domain: "automaton",
+            evt: "ended",
+            automatonType: "satisfactionSurvey",
+            siteID: Number(chatParams.siteID),
+            customerID: chatParams.thisCustomerID,
+            incAssignmentID: chatParams.sessionID,
+            pageID: Number(chatParams.launchPageId),
+            sessionID: chatParams.sessionID,
+            chatID: chatParams.chatID,
+            preAssigned: this.sdk.isConnected() && !chatParams.agentID,
+            automatonID: automaton.id,
+            "automaton.outcomeType": "Refused",
+            clientTimestamp: timestamp,
+            automatonOrigin: "richMedia"
+        };
+        console.log("===== closePostChatSurvey =====");
+
+        try {
+            this.sdk.logEventToDW({ eventList: [endedEvent] });
+        } catch (e) {
+            console.error("!!!! logEventToDW got exception: ", e);
+        }
+    }
+}
```

Comment:
Can we add New line at the end of the file please?

Reply:
Done